### PR TITLE
[Feature] Toggle Pokemon cries (during battle)

### PR DESCRIPTION
**Pull request by flx-sta**
## What are the changes the user will see?

A new setting will be available to turn PKM cries on/off during battle. 
> [!WARNING]
> This will not affect cries in the starter-select screen or in the evolution phase!

## Why am I making these changes?
It's a highly requested feature and can also speed up tests

## What are the changes from a developer perspective?
The `BattleScene` has a new field called `enablePokemonCries` (Same old...).
This fields value is checked during `faintCry()`, `encounter-phase` and `summon-phase`.
Some callback delay values are automatically adjusted too as we don't want the player to have to wait for a cry that isn't even being played.

### Screenshots/Videos

image

## How to test the changes?

- Adjust the setting to on or off
- Play 1 wave and make sure that both your pokemon and the enemy pokemon is fainted to check if the faint cries are also (not) playing

## Checklist
- [x] **I'm using `beta` as my base branch**
- [x] There is no overlap with another PR?
- [x] The PR is self-contained and cannot be split into smaller PRs?
- [x] Have I provided a clear explanation of the changes?
- ~~[ ] Have I considered writing automated tests for the issue?~~
- [x] If I have text, did I make it translatable and add a key in the English locale file(s)?
- [x] Have I tested the changes (manually)?
    - [x] Are all unit tests still passing? (`npm run test`)
- [x] Are the changes visual?
  - [x] Have I provided screenshots/videos of the changes?

### CODE REVIEW

**Review by Adri1 — DISMISSED** [review]

File: src/locales/en/settings.json [hunk 1]
```diff
@@ -103,5 +103,6 @@
   "rewards": "Rewards",
   "reroll": "Reroll",
   "shop": "Shop",
-  "checkTeam": "Check Team"
+  "checkTeam": "Check Team",
+  "pokemonCries": "Pokemon Cries"
```
Comment (Adri1) [inline src/locales/en/settings.json:107]:
```suggestion
  "pokemonCries": "Pokémon Cries"
```

**Review by Adri1 — APPROVED** [review]

**Review by frutescens — CHANGES_REQUESTED** [review]
Wouldn't it make more sense to go to the function cry() in pokemon-species.ts and disable the cry there? This way, cries will be disabled at the source. Also, wouldn't it be better to check / set scene.fieldVolume to 0 and use that in pokemon-species.ts' cry() instead of creating a new setting?

**Comment by frutescens** [comment]
Mm, I did take a look at faintCry() and fusionFaintCry() and they do seem fairly clunky. It is surprising that they're very different from cry() but I do get why it's not so easily cut and dry.

**Comment by DayKev** [comment]
If it only affects cries during battle and not elsewhere (ie: evolution) then maybe the label in the settings should be "Pokémon Cries In Battle [y/n]"

**Comment by DayKev** [comment]
Btw I checked the test speed-up vs beta and:

Before:
image
After:
image

Not incredible or anything, but it's decent. However, when one or more Pokémon faint:

Before:
image
After:
image

Insane. `FaintPhase` no longer takes multiple whole seconds (!).

**Comment by flx-sta** [comment]
> Wouldn't it make more sense to go to the function cry() in pokemon-species.ts and disable the cry there? This way, cries will be disabled at the source. Also, wouldn't it be better to check / set scene.fieldVolume to 0 and use that in pokemon-species.ts' cry() instead of creating a new setting?

This might be something to consider… yeah.
Is field only cries and faint cries anyway ?

regarding not changing `cry()`: cause I didn’t want it to affect evolutions & starter select cry’s. But if that is affected by “field” volume too I would probably cut it out too

**Comment by frutescens** [comment]
Yes, field volume covers all cries + battle sounds. I think this feature is useful for devs, but not so useful for users (although I could see performance arguments for putting similar checks for other sound cues like BGM)

**Comment by flx-sta** [comment]
@frutescens may take over this feature

**Pull request closed without merging**